pkg/keys/sql.go

```diff
@@ -259,3 +259,35 @@ func (d sqlDecoder) DecodeTenantMetadataID(key roachpb.Key) (roachpb.TenantID, e
 	}
 	return roachpb.MakeTenantID(id), nil
 }
+
+// RewriteSpanToTenantPrefix updates the passed Span, potentially in-place, to
+// ensure the Key and EndKey have the passed tenant prefix, regardless of what
+// prior tenant prefix, if any, they had before, and returns the updated Span.
+func RewriteSpanToTenantPrefix(sp roachpb.Span, prefix roachpb.Key) (roachpb.Span, error) {
+	var err error
+	sp.Key, err = rewriteKeyToTenantPrefix(sp.Key, prefix)
+	if err != nil {
+		return sp, err
+	}
+	sp.EndKey, err = rewriteKeyToTenantPrefix(sp.EndKey, prefix)
+	return sp, err
+}
+
+func rewriteKeyToTenantPrefix(key roachpb.Key, prefix roachpb.Key) (roachpb.Key, error) {
+	if len(prefix) == 0 && bytes.HasPrefix(key, TenantPrefix) || !bytes.HasPrefix(key, prefix) {
+		suffix, _, err := DecodeTenantPrefix(key)
+		if err != nil {
+			return nil, err
+		}
+
+		if extra := len(key) - len(prefix) - len(suffix); extra >= 0 {
+			key = key[extra:]
+			copy(key, prefix)
+		} else {
+			key = make(roachpb.Key, len(prefix)+len(suffix))
```

Comment:
I checked out the PR, and I don't think we are testing this branch in the logictest. Are we able to run logictests inside a tenant yet? Thats the only way I can see us testing this, where the key would have a shorter prefix, than the passed in `prefix`?

Reply:
Added a unit test, new cover profile: <img width="696" alt="Screen Shot 2022-01-18 at 7 58 52 PM" src="https://user-images.githubusercontent.com/15615/150043777-93742741-787d-4cc1-8842-31612827ff0c.png">

Reply:
Neat, thanks!